compiler/parser.mly

```diff
@@ -106,6 +106,7 @@ assignment_data_type:
   | FLOAT   { "float" }
   | STRING  { "string" }
   | JSON    { "json" }
+  /* What happened to booleans? */
```

Comment:
Hardcoded it for bool: 'BOOL ID ASSIGN bool_expr', since we are assigning it to a bool_expr.